python/nano/src/bigdl/nano/tf/keras/customized_training_utils.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import tensorflow as tf
+import numpy as np
+from functools import wraps
+
+
+def tf_bf16_nano(func):
+    """A decorator to realize mixed precision on customized training loop."""
+    # todo check the func signature
+    @wraps(func)
+    def wrapper(*inner_args):
+        new_args = []
+        for arg in inner_args:
+            if isinstance(arg, (tf.Tensor, np.ndarray)):
```

Comment:
This may only work as temporary solution. A case that may break this is
1. model would give a fp32 result no matter what the policy is (https://www.tensorflow.org/guide/mixed_precision?hl=zh-cn#%E6%9E%84%E5%BB%BA%E6%A8%A1%E5%9E%8B)

Reply:
> 

Indeed, we can improve this solution in future work.